Baekjoon/2231_1.py

```diff
@@ -2,25 +2,38 @@
 문제 링크:https://www.acmicpc.net/problem/2231
 카테고리:브루트포스
 문제해설:분해합의 생성자 구하기
+
+M의 각자리수 + M
+
+*523%10=3 append->(523-3)//10=52
+->*52%10=2 append-> (52-2)//10=5=b
+->if b<10->break
+
+216입력->198출력
 '''
-n=int(input())
 
-def func(m):
+N=int(input())
+
+def M_sum(M):
   arr=[]
-  a=m%10
-  arr.append(m)
-  m=(m-a)%10
-  print(arr)
-  if m!=0:
-    func(m)
-  elif m==0:
-    m=(m-a)//10
-    arr.append(m)
-
-for m in range(1,n+1):
-  m_sum=m+sum(func(m))
-  if m_sum==n:
-    print(m)
-    break
-  else:
-    print(0)
+
+  while True:
+    a=M%10
+    arr.append(a)
+    M=(M-a)//10
+  
+    if M<10:
+      arr.append(M)
+      break
+
+  return sum(arr)
+def result_M(N):
+  
+  for M in range(N+1):
+    M_sum_all=M+M_sum(M)
+    if M_sum_all==N:
+      return M
```

Comment:
```suggestion
  for M in range(N+1):
    if M_sum(M)==N:
      return M
```

M_sum(M) 함수를 M + M_sum(M) 과 같은 결과값을 반환하도록 하면 L33이 없어질 수 있겠습니다.